File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        Map<Character, Integer> charMap = new HashMap<>();
        int len = s.length();
        int[] answer = new int[len];
        Arrays.fill(answer, -1);
        
        for (int i = 0; i < len; i++) {
            char c = s.charAt(i);
            if (charMap.containsKey(c)) {
                answer[i] = i - charMap.get(c);
            }
            charMap.put(c, i);
        }
        
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.kt
class Solution {
    fun solution(s: String): IntArray {
        val charMap = mutableMapOf<Char, Int>()
        var answer = IntArray(s.length) { -1 }
        
        for (i in s.indices) {
            val c = s[i]
            val preIdx = charMap.getOrDefault(c, -1)
            
            if (preIdx != -1) {
                answer[i] = i - preIdx
            }
            
            charMap[c] = i
        }
        
        return answer
    }
}